### PR TITLE
[TASK] Add section about how to get the PSR-7 request object

### PR DESCRIPTION
**Pull request by brotkrueml**
More to come:
- Add new attributes from TYPO3 v12

Releases: main, 11.5
Related: #2337

### CODE REVIEW

File: Documentation/ApiOverview/RequestLifeCycle/Typo3Request.rst [hunk 1]
```diff
@@ -11,14 +11,136 @@ TYPO3 request object
...
+    {
+        // ...
+
+        // Retrieve the language attribute via the request object
+        $language = $this->request->getAttribute('language');
+
+        // ...
+    }
+
+Prior to TYPO3 v11.3, a custom Extbase request object is available that does not
```
Comment (linawolf) [inline Documentation/ApiOverview/RequestLifeCycle/Typo3Request.rst:54]: Shall we still document that here? This information would belong in the 10.4 docs?

File: Documentation/ApiOverview/RequestLifeCycle/Typo3Request.rst [hunk 2]
```diff
@@ -11,14 +11,136 @@ TYPO3 request object
...
+
+The request object compatible with the PSR-7
+:php:`\Psr\Http\Message\ServerRequestInterface` is available in an
+:ref:`Extbase controller <extbase-action-controller>` via the class property
+:php:`$this->request`:
+
+..  code-block:: php
+    :caption: EXT:my_extension/Classes/Controller/MyController.php
+
+    public function myAction(): ServerResponseInterface
```
Comment (linawolf) [inline Documentation/ApiOverview/RequestLifeCycle/Typo3Request.rst:44]: I would keep the class declaration in all of these examples, as the class that is being implented shows the context we are in without reading the text above. Also please keep the include statements (yes it is repetition about theServerResponseInterface but still usefull to new users )

File: Documentation/ApiOverview/RequestLifeCycle/Typo3Request.rst [hunk 3]
```diff
@@ -11,14 +11,136 @@ TYPO3 request object
...
+
+        // ...
+    }
+
+
+Data processor
+--------------
+
+A :ref:`data processor <content-elements-custom-data-processor>` receives a
+reference to the :php:`ContentObjectRenderer` as first argument for the
```
Comment (linawolf) [inline Documentation/ApiOverview/RequestLifeCycle/Typo3Request.rst:87]: please add FQN

File: Documentation/ApiOverview/RequestLifeCycle/Typo3Request.rst (hunk 1 above)
Comment (brotkrueml) [reply Documentation/ApiOverview/RequestLifeCycle/Typo3Request.rst:54]: When having an extension compatible with 10.4 and 11.5 it does matter (change is merged into 11.5). We can remove it later for the v12 docs.

Comment (linawolf) [reply Documentation/ApiOverview/RequestLifeCycle/Typo3Request.rst:54]: Mmh, then i would wirite what to do to be compatible with v11 and v10

Comment (brotkrueml) [reply Documentation/ApiOverview/RequestLifeCycle/Typo3Request.rst:54]: Agreed.

**Review by linawolf — APPROVED** [review]

**Pull request merged**